src/tcgwars/logic/impl/gen3/PowerKeepers.groovy

```diff
@@ -1608,11 +1608,9 @@ public enum PowerKeepers implements LogicCardInfo {
           energyCost W
           attackRequirement {}
```

Comment:
This attack should have 2 requirements: bench not full and deck not empty, it's better to put these in the attackRequirement clause like this: 

```suggestion
          attackRequirement {
              assert my.bench.notFull : "Bench is full"
              assert my.deck : "Deck is empty"
          }
```

and remove `if(my.deck)` inside `onAttack`.